### DISCUSSION TITLE
How do you return a (custom ?) error page with 410 status code inside getStaticProps ?

### DISCUSSION THREAD

**Discussion by driolet** [vercel/next.js#41257] [Help]
Hello, 

I have a question and I don't know this is actually doable, that's why I'm here. 

On my product page, i'm using getStaticPaths and getStaticProps functions for getting API (from API Platform) product data. 

The thing is when I don't have any data for the researched product, I use the `notFound`  from NextJS (which return 404 page and status code).

```
export async function getStaticProps({ params }) {
 
  try {
    const res = await ApiService.getUniqueBien(params.affNum);
    if(!res)
      return {
        // returns the default 404 page with a status code of 404
        notFound: true,
      };
      
    [...]

```

However, when a product is not available anymore, I want to use the [410](https://httpwg.org/specs/rfc9110.html#status.410) status code instead of 404 (Google continues to index my product pages which is problematic).  My 404 page is already a custom page and I want to reuse the same style on it.

How can I do it (if this is possible of course) ?

Have a great day, 
Best regards,

**Comment by gaston-flores** [comment]
I don't think this is currently possible with `getStaticProps`. You can change the status code with `getServerSideProps`, but you'll lose the benefits of static rendering.

**Comment by rsheppard-dev** [comment]
Status codes are generated on the server so as said I don’t think this is possible.

Instead of returning notFound you could return  a prop to tell your main page no product was found and just return a 410 component instead of product page.

**Reply by driolet** [reply]
Thank you for your fast answer, 

I already tried returning a prop and return a custom component that but I didn't manage to trigger an 4xx error. 
I only trigger a 200 response, which is not what I wanted (my 404 page already explain than the product may be unavailable).

Is it possible to trigger an 4xx error in a component without having a 200 response on the main page ?

**Reply by 240p60** [reply]
I must do the same. Have you found a solution?

`context.res.statusCode = 410;`
I caused a 410 error, but in this case, I can't get my custom error page.
